salt/modules/riak.py

```diff
@@ -44,10 +59,20 @@ def stop():
 
         salt '*' riak.stop
     '''
-    return not bool(__salt__['cmd.retcode']('riak stop'))
+    ret = {'comment': '', 'success': False}
 
+    cmd = __execute_cmd('riak', 'stop')
 
-def cluster_join(riak_user=None, riak_host=None):
+    if cmd['retcode'] != 0:
+        ret['comment'] = cmd['stderr']
+    else:
+        ret['comment'] = cmd['stdout']
+        ret['success'] = True
+
+    return ret
+
+
+def cluster_join(username, hostname):
```

Comment:
Can you add some `versionchanged` tags here for both of the args that you changed? Can you add some docs for both of these args as well?